test/integration/tlspolicy_controller_test.go

```diff
@@ -80,13 +81,49 @@ var _ = Describe("TLSPolicy", Ordered, func() {
 
 		AfterEach(func() {
 			err := k8sClient.Delete(ctx, gateway)
-			Expect(err).ToNot(HaveOccurred())
-			err = k8sClient.Delete(ctx, tlsPolicy)
-			Expect(err).ToNot(HaveOccurred())
+			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
 		})
 
-		Context("valid target, issuer and policy", func() {
+		Context("gateway with no TLS Policy and multiple listeners", func() {
+			manualSecretName := "manual-tls-secret"
+			manualListenerName := "manual-test.example.com"
+			BeforeEach(func() {
+				gateway = NewTestGateway("test-gateway", gwClassName, testNamespace).
+					WithHTTPSListener(manualListenerName, manualSecretName).
+					WithHTTPSListener("test2.example.com", "test2-tls-secret").
+					WithHTTPSListener("*.example.com", "wildcard-test-tls-secret").Gateway
+				Expect(k8sClient.Create(ctx, gateway)).To(BeNil())
+				Eventually(func() error { //gateway exists
+					return k8sClient.Get(ctx, client.ObjectKey{Name: gateway.Name, Namespace: gateway.Namespace}, gateway)
+				}, TestTimeoutMedium, TestRetryIntervalMedium).ShouldNot(HaveOccurred())
+			})
+			AfterEach(func() {
+				err := k8sClient.Delete(ctx, gateway)
+				Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
+			})
+			It("should not be programmed", func() {
+				Consistently(func() error {
+					freshGW := &gatewayv1beta1.Gateway{}
+					err := k8sClient.Get(ctx, client.ObjectKey{Name: gateway.Name, Namespace: gateway.Namespace}, gateway)
+					if err != nil {
+						return err
+					}
+					if freshGW.Status.Conditions == nil {
+						return nil
+					}
+					for _, condition := range freshGW.Status.Conditions {
+						if condition.Type == string(gatewayv1beta1.GatewayConditionProgrammed) {
+							if strings.ToLower(string(condition.Status)) == "true" {
+								return fmt.Errorf("expected programmed status false, got true")
+							}
+						}
+					}
```

Comment:
Why do you get a gateway and update `gateway`, but check the conditions on  `freshGW`. What is this test testing?

Reply:
bah, this is wrong. I'll fix it now thanks.

Reply:
sorted now.